be/src/olap/rowset/beta_rowset_reader.cpp

```diff
@@ -269,6 +279,10 @@ Status BetaRowsetReader::next_block(vectorized::Block* block) {
         if (!s.ok()) {
             if (!s.is<END_OF_FILE>()) {
                 LOG(WARNING) << "failed to read next block: " << s.to_string();
+            } else {
+                if (_iterator->is_borrowed()) {
```

Comment:
`_iterator->is_borrowed()` seems to always be false, `_iterator` is `VMergeIterator` or `VUnionIterator`

Is it supposed to be like this?
```
for (RowwiseIteratorUPtr& it : _origin_iters) {
             if (it && it->is_borrowed()) {
                 ResourcePool::return_iterator(std::move(it));
             }
         }
```

Reply:
since _iter may be SegmentIterator when rowset only contains a single segment